### DISCUSSION TITLE
How can import pages defined in a npm package?

### DISCUSSION THREAD

**Discussion by andreazanti** [vercel/next.js#21197] [Help] [answered]
Now i have my pages under /pages with the two methods getStaticPaths and getStaticProps. All is ok. 
Due to business model i need to have the pages under a npm package.

Something like this:

 **pages/home.tsx**
```

import {initHome} from "myPackage"

export default initHome(someParamsHere);
```

myPackage
```

// Here the Home component + getStaticProps + getStaticPaths

export initHome = () => {
     return <Home></Home>
}

```

How can i replicate this behaviour?

**Comment by vassbence** [comment, marked as answer]
Next.js only supports `getStaticProps` and similar functions when they are exported from files in the `pages` folder. What you could do is have a function like `fetchPageData` in your package, and do an
```js
import { HomePage, fetchHomePageData } from "myPackage"

export async function getStaticProps(params) {
  return fetchHomePageData(params)
}

export default HomePage
```

Make sure that `fetchPageData` returns data in the correct format

**Reply by andreazanti** [reply]
Ok understood, thx!
